lstchain/visualization/plot_dl2.py

```diff
@@ -345,38 +337,116 @@ def plot_importances(clf,features):
+        print("%d. %s (%f)" % (f + 1,
+                               features[indices[f]],
                                importances[indices[f]]))
 
     ordered_features=[]
     for index in indices:
         ordered_features=ordered_features+[features[index]]
-    
-    plt.title("Feature importances for G/H separation",
+
+    plt.title("Feature importances",
               fontsize=15)
-    plt.bar(range(len(features)), 
+    plt.bar(range(len(features)),
             importances[indices],
             color="r", yerr=std[indices], align="center")
-    plt.xticks(range(len(features)), 
+    plt.xticks(range(len(features)),
                ordered_features)
-    plt.xlim([-1, 
+    plt.xlim([-1,
               len(features)])
 
 def plot_ROC(clf,data,features, Energy_cut):
     # Plot ROC curve:
     check = clf.predict_proba(data[features])[0:,1]
-    accuracy = accuracy_score(data['hadroness'], 
+    accuracy = accuracy_score(data['hadroness'],
                               data['hadro_rec'])
     print(accuracy)
-    
+
     fpr_rf, tpr_rf, _ = roc_curve(data['hadroness'],
                                   check)
-    
-    plt.plot(fpr_rf, tpr_rf, 
+
+    plt.plot(fpr_rf, tpr_rf,
              label='Energy Cut: '+'%.3f'%(pow(10,Energy_cut)/1000)+' TeV')
     plt.xlabel('False positive rate',
                fontsize=15)
     plt.ylabel('True positive rate',
                fontsize=15)
     plt.legend(loc='best')
+
+def plot_e_resolution(data, Nbins, emin, emax):
+
+    plt.rcParams['figure.figsize'] = (30, 10)
+    plt.rcParams['font.size'] = 14
+
+    #difE = ((data['mc_energy']-data['e_rec'])*np.log(10))
+    difE = np.log(10**data['e_rec']/10**data['mc_energy'])
+    means_result = scipy.stats.binned_statistic(
+        data['mc_energy'],[difE,difE**2],
+        bins=Nbins,range=(emin, emax),statistic='mean')
+    means, means2 = means_result.statistic
+    standard_deviations = np.sqrt(means2 - means**2)
+    bin_edges = means_result.bin_edges
+    bin_centers = (bin_edges[:-1] + bin_edges[1:])/2.
+
+    gs0 = gridspec.GridSpec(1,2,width_ratios=[1,2])
+    subplot = plt.subplot(gs0[0])
+    gs = gridspec.GridSpecFromSubplotSpec(2, 1,
+                                          height_ratios=[1, 1],
+                                          subplot_spec=subplot)
+
+    ax0 = plt.subplot(gs[0])
+    plot0 = ax0.errorbar(x=bin_centers,
+                         y=means, yerr=standard_deviations,
+                         linestyle='none', marker='.')
+
+    plt.ylabel('Bias',fontsize=24)
+    plt.grid()
+    ax1 = plt.subplot(gs[1],sharex = ax0)
+    plot1 = ax1.plot(bin_centers,standard_deviations,
+                     marker='X',linestyle='None')
+    plt.ylabel('STD',fontsize=24)
+    plt.xlabel('$log_{10}E_{true}(GeV)$',fontsize=24)
+    plt.grid()
+
+    subplot2 = plt.subplot(gs0[1])
+
+    #Lines for setting the configuration of the subplots depending on Nbins
+    import math
```

Comment:
Do not import inside the function.
Use `np.ceil`.

Reply:
Changed! It's a residual from having these functions in a messy notebook...